### PR TITLE
Logout pixabay

### PR DESCRIPTION
**Pull request by janegreene**
### New Feature Submissions:
* [x] Have you added an explanation of what your changes do?
* [x] Have you linted your code locally prior to submission?
I have one overly long method that will need to be addressed in refactor. 
### Testing:
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?
* [x] All new and existing tests passed?

**Test coverage** =100%
--------------------------------------------------------------------------------
### Change Log:
Created routes and view pages for 400, 404, and 500 errors. 
Logout button is now functional. 
Happy path for two word searches and pixabay images.
Sad path handling for three word search and non-alpha characters for both photo APIs.

### CODE REVIEW

File: app/controllers/search_controller.rb [hunk 1]
```diff
@@ -1,9 +1,18 @@
 class SearchController < ApplicationController
   before_action :require_user
   def index
```
Comment (janegreene) [inline app/controllers/search_controller.rb:3]: This method is in need of helpers or shortening in some way.

File: app/controllers/search_controller.rb [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
-    conn = Faraday.get("https://damp-forest-93176.herokuapp.com/images/#{params[:search]}")
-    json = JSON.parse(conn.body, symbolize_names: true)
-    @image_options = ImageOptions.new(json)
+    search = params[:search].split.join('+')
+    if search.match('[a-zA-Z]+')
```
Comment (janegreene) [inline app/controllers/search_controller.rb:5]: This line confirms all characters are letters or spaces replaced by '+' for the sake of the API searches.

File: app/controllers/search_controller.rb [hunk 3]
```diff
@@ -1,9 +1,18 @@
...
+      conn = Faraday.get("https://damp-forest-93176.herokuapp.com/images/#{search}")
+      if conn.status == 500
+        redirect_to '/500'
+      elsif conn.status == 200
+        json = JSON.parse(conn.body, symbolize_names: true)
+        @image_options = ImageOptions.new(json)
+      end
+    else
+      redirect_to '/400'
```
Comment (janegreene) [inline app/controllers/search_controller.rb:14]: This is happening when we search gibberish or special characters that the API does not enjoy.

File: config/routes.rb [hunk 4]
```diff
@@ -4,9 +4,12 @@
...
   get '/login', to: 'sessions#new'
-  get '/logout', to: 'sessions#destroy'
+  delete '/logout', to: 'sessions#destroy'
   get '/dashboard', to: 'users#show'
   get '/search', to: 'search#index'
   get '/search/:id', to: 'search#show', as: :search_show
   post '/image', to: 'images#create'
+  get '/500', to: 'application#server_errors'
+  get '/400', to: 'application#search_errors'
+  get '(*url)', to: redirect('/404')
```
Comment (janegreene) [inline config/routes.rb:14]: This protects from the app errors when a random path is attempted by the user.

File: app/controllers/search_controller.rb (hunk 3 above)
Comment (kathytbui) [reply app/controllers/search_controller.rb:14]: I think this method could be refactored out if we end up having time to work on that.

File: config/routes.rb (hunk 4 above)
Comment (kathytbui) [reply config/routes.rb:14]: Does this cover all random hardcoded urls that we don't have routes for?

Comment (dtsehl) [reply config/routes.rb:14]: That's what it should do, yes.

File: spec/features/dashboard/show_spec.rb [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 RSpec.describe 'Dashboard' do
   describe 'As an authenticated of user' do
-    it "text" do
+    it "can search with keywords" do
```
Comment (dtsehl) [inline spec/features/dashboard/show_spec.rb:5]: Good catch on changing this typo.

**Review by dtsehl — APPROVED** [review]

File: public/400.html [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+  @import url(https://fonts.googleapis.com/css?family=Gilda+Display);
+
+  html {
+    background: radial-gradient(#000, #111);
+    color: white;
+    overflow: hidden;
+    height: 100%;
+    user-select: none;
+  }
+
```
Comment (sciencefixion) [inline public/400.html:16]: This css is intense!

**Review by sciencefixion — APPROVED** [review]
Fantastic!

**Pull request merged**